plugin/data_masking/src/udf/udf_mask_ssn.cc

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2018, 2019 Francisco Miguel Biete Banon. All rights reserved.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software Foundation,
+   51 Franklin Street, Suite 500, Boston, MA 02110-1335 USA */
+
+#include "plugin/data_masking/include/udf/udf_mask_ssn.h"
+#include "plugin/data_masking/include/plugin.h"
+#include "plugin/data_masking/include/udf/udf_utils.h"
+#include "plugin/data_masking/include/udf/udf_utils_string.h"
+
+static bool mask_ssn_init(UDF_INIT *initid, UDF_ARGS *args, char *message) {
+  DBUG_ENTER("mask_ssn_init");
+
+  if (args->arg_count != 1) {
+    std::snprintf(message, MYSQL_ERRMSG_SIZE,
+                  "Wrong argument list: mask_ssn(string)");
+    DBUG_RETURN(true);
+  }
+
+  if (args->arg_type[0] != STRING_RESULT) {
+    std::snprintf(message, MYSQL_ERRMSG_SIZE,
+                  "Wrong argument type: mask_ssn(string)");
+    DBUG_RETURN(true);
+  }
+
+  initid->maybe_null = 1;
+  initid->ptr = NULL;
+
+  DBUG_RETURN(false);
+}
+
+static void mask_ssn_deinit(UDF_INIT *initid) {
+  DBUG_ENTER("mask_ssn_deinit");
+
+  if (initid->ptr) free(initid->ptr);
```

Comment:
memory allocated with new(), do not use free().

Just use  delete initd->ptr

Reply:
Thank you, done!